File: deepseek/2.Estructura de Datos Pila (Stack)/golang/pila.go
package main

import (
	"fmt"
	"strings"
)

type Stack struct {
	items []string
}

func (s *Stack) Push(item string) {
	s.items = append(s.items, item)
}

func (s *Stack) Pop() string {
	if len(s.items) == 0 {
		return ""
	}
	item := s.items[len(s.items)-1]
	s.items = s.items[:len(s.items)-1]
	return item
}

func (s *Stack) IsEmpty() bool {
	return len(s.items) == 0
}

func main() {
	stack := Stack{}
	for {
		fmt.Print("¿Desea 'push', 'pop' o 'salir'? ")
		var action string
		fmt.Scanln(&action)
		action = strings.ToLower(action)
		if action == "salir" {
			break
		} else if action == "push" {
			fmt.Print("Ingrese el valor a apilar: ")
			var item string
			fmt.Scanln(&item)
			stack.Push(item)
		} else if action == "pop" {
			popped := stack.Pop()
			if popped != "" {
				fmt.Println("Elemento desapilado:", popped)
			} else {
				fmt.Println("La pila está vacía")
			}
		}
	}
}

File: deepseek/2.Estructura de Datos Pila (Stack)/Typescript/pila.ts
class NumberStack<T> {
    private items: T[] = [];

    push(item: T): void {
        this.items.push(item);
    }

    pop(): T | undefined {
        return this.items.pop();
    }

    isEmpty(): boolean {
        return this.items.length === 0;
    }

    print(): void {
        console.log("Pila actual:", this.items);
    }
}

// Interacción con el usuario
const numberStack = new NumberStack<number>();
while (true) {
    const action = prompt("¿Qué desea hacer? (push/pop/salir):")?.toLowerCase();
    
    if (action === 'salir') break;
    
    switch (action) {
        case 'push':
            const value = parseInt(prompt("Ingrese un número:") || "0");
            numberStack.push(value);
            break;
        case 'pop':
            const popped = numberStack.pop();
            console.log("Elemento eliminado:", popped);
            break;
        default:
            console.log("Acción no válida");
    }
    numberStack.print();
}